01_fib/pokkalar/main.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func fib(n int) {
+	sign, prev, curr := 1, 0, 1
+	if n < 0 {
+		sign = -1
+	}
+	for i := 0; i < n*sign; i++ {
+		fmt.Fprintln(out, curr)
+		sum := prev + curr*sign
```

Comment:
Declaration of sum helps in better readability, but it may not be required in this case.

prev , curr = curr, prev + curr*sign.

But its understandable if you wish to keep sum.

Reply:
Thanks,I will keep the sum variable

Reply:
fwiw, I would have also chosen the one liner....